### PR TITLE
fix: tidy up the project

### PR DESCRIPTION
**Pull request by Decodetalkers**
use applicationManager instead of Launcherd1 since that interface is moved
update the usage of cmake, use scanner instead
mask dmemorywarningdialog since startmanager is removed update cmake to c++17

Log:

### CODE REVIEW

**Comment by Decodetalkers** [comment]
- https://github.com/linuxdeepin/developer-center/issues/6792

File: dmemory-warning-dialog/src/dmemorywarningdialog.cpp [hunk 1]
```diff
@@ -49,7 +49,7 @@ bool confirm(const QPixmap &icon)
 DMemoryWarningDialog::DMemoryWarningDialog(QWidget *parent)
     : DAbstractDialog(parent)
     // TODO
-    , m_startManagerInter(new StartManagerInter("org.deepin.dde.StartManager1", "/org/deepin/dde/StartManager1", QDBusConnection::sessionBus(), this))
+    //, m_startManagerInter(new StartManagerInter("org.deepin.dde.StartManager1", "/org/deepin/dde/StartManager1", QDBusConnection::sessionBus(), this))
```
Comment (BLumia) [inline dmemory-warning-dialog/src/dmemorywarningdialog.cpp:52]: 留着还有用么？直接删掉吧？下同

Comment (Decodetalkers) [reply dmemory-warning-dialog/src/dmemorywarningdialog.cpp:52]: 这个地方是我不知道用什么替代了，这个工具似乎没有很重要，但是这个地方需要修复，所以这边加个todo吧

**Review by ComixHe — APPROVED** [review]

**Review by BLumia — APPROVED** [review]

**Pull request merged**